File: java/web_tutorial_mastercopy/tut_tree_p1.java
import bridges.connect.Bridges;
import bridges.base.TreeElement;

public class tree {
	public static void main(String[] args) throws Exception {


		//create the Bridges object
#if TESTING
		Bridges bridges = new Bridges(Integer.parseInt(args[0]), args[1], args[2]);
		bridges.setServer(args[3]);


#else
		Bridges bridges = new Bridges(YOUR_ASSIGNMENT_NUMBER, "YOUR_USER_ID",
			"YOUR_API_KEY");
#endif

		// title, description
		bridges.setTitle("A General Tree Example");
		bridges.setDescription("A basic tree with seven nodes. Three on one side and three on the other. The root node is set to red with 0.3 opacity. "
			+	"The other six nodes are neutral color.");

		//create  tree nodes
		TreeElement<String> t0 = new TreeElement<String>("Hello", "Hello");
		TreeElement<String> t2 = new TreeElement<String>("This", "This");
		TreeElement<String> t3 = new TreeElement<String>("is", "is");
		TreeElement<String> t4 = new TreeElement<String>("a", "a");
		TreeElement<String> t5 = new TreeElement<String>("generic", "generic");
		TreeElement<String> t6 = new TreeElement<String>("Tree", "Tree");
		TreeElement<String> t7 = new TreeElement<String>("Representation", "Representation");

		// add links to children
		t0.addChild(t2);
		t0.addChild(t3);
		t0.addChild(t4);
		t2.addChild(t5);
		t2.addChild(t6);
		t3.addChild(t7);

		bridges.setTitle("A General Tree Example");

		// visualize the tree
		bridges.setDataStructure(t0);
		bridges.visualize();
	}
}

File: python/web_tutorial_mastercopy/tut_tree_p1.py
from bridges.bridges import *
from bridges.tree_element import *
import sys

def main():
    # create the Bridges object, set credentials
#if TESTING
    # command line args provide credentials and server to test on
    args = sys.argv[1:]
    bridges = Bridges(int(args[0]), args[1], args[2])
    if len(args) > 3:
        bridges.connector.set_server(args[3])
#else
    bridges = Bridges(YOUR_ASSSIGNMENT_NUMBER, "YOUR_USER_ID", "YOUR_API_KEY")
#endif

    # title, description
    bridges.set_title("A General Tree Example")
    bridges.set_description("A basic tree with seven nodes. Three on one side and three on the other. The root node is set to red with 0.3 opacity. The other six nodes are neutral color.")

    # create tree nodes
    t0 = TreeElement(e="Hello", label="Hello")
    t2 = TreeElement(e="this", label="this")
    t3 = TreeElement(e="is", label='is')
    t4 = TreeElement(e="a", label='a')
    t5 = TreeElement(e="generic", label="generic")
    t6 = TreeElement(e="tree", label='tree')
    t7 = TreeElement(e="representation", label="representation")

    # add links to children
    t0.add_child(t2)
    t0.add_child(t3)
    t0.add_child(t4)
    t2.add_child(t5)
    t2.add_child(t6)
    t3.add_child(t7)

    # set visualizer type
    bridges.set_data_structure(t0)

    # visualize the tree
    bridges.visualize()


if __name__ == "__main__":
    main()

